### PR TITLE
Revise rt.sh to create new baseline automatically.

### LINKED ISSUE #32: Revise rt.sh to create new baseline automatically
The shell script "rt.sh" needs to be revised so that new baseline can be created automatically using the following command:
./rt.sh -c -k -f

Task dependency feature needs to be added. A warm-start run will start only after the corresponding cold-start run has been completed.

### PR DESCRIPTION
**Pull request by binli2337**
1. The script "rt.sh" is revised so that new baseline can be created automatically.
2. Task dependency feature is added to the regression test system.

close #32

### CODE REVIEW

**Review by junwang-noaa — COMMENTED** [review]
Thanks for making this working. I assume we still need two steps in order to run warm cases, is that correct?

**Comment by binli2337** [comment]
Yes. We still need two steps. The new mediator restart files need to be copied from the new baseline directory to the original baseline directory.

**Review by junwang-noaa — APPROVED** [review]

**Comment by DeniseWorthen** [comment]
I don't understand what you mean by "creating the baseline automatically". Could you explain what issue this PR fixes?

**Comment by binli2337** [comment]
There are two ways to create a new baseline:

1. Creating a new baseline manually
This can be done by running
./rt.sh -k
After the tests are completed, you can create a new baseline manually:
a) Creating a new baseline directory manually and copying all fixed files from the most recent baseline directory to this directory.
b) Copying regression test results from your run directory to the new baseline directory.

2. Creating a new baseline automatically
This can be done by running
./rt.sh -c -k
After the tests are completed, a new baseline directory is created. The new directory contains both fixed files and regression test results.

**Comment by DeniseWorthen** [comment]
So this replicates the baseline creation functionality of s2s but does not include the dependency feature, correct?

**Comment by binli2337** [comment]
Yes.

**Comment by junwang-noaa** [comment]
Bin,

Does this mean you are using mediator flux from previous baseline, not from
the current cold start run for the 2day run?

**Comment by binli2337** [comment]
For the 2-day runs, the mediator restart files are from the previous baseline area since the dependency feature has not been implemented in DATM-MOM6-CICE5 yet.

**Comment by DeniseWorthen** [comment]
I'm confused. If you're creating a new baseline, my understanding is first you create the mediator restart files for the new baseline by running the cold tests. Then you create the new baseline for the warm tests using those new files. It is two separate steps, but you need new mediator files if you have a new baseline, right?

**Comment by binli2337** [comment]
The test cases have been revised. When CREATE_BASELINE=true, the 2-day runs will use mediator restart files from the new baseline rather than the previous baseline.

File: tests/datm_conf/cpld_datm_mom6_cice5_cfsr.IN [hunk 1]
```diff
@@ -20,5 +20,9 @@ cp    @[RTPWD]/CICE_IC/cice5_model_0.25.cpc*.nc cice5_model.res.nc
 
 # WARM START
 if [[ ! $MED_restart_data == '' ]]; then
+  if [[ @[CREATE_BASELINE] == 'true' ]]; then
+  cp  @[NEW_BASELINE]/MEDIATOR_CFSR/mediator_* .
```
Comment (junwang-noaa) [inline tests/datm_conf/cpld_datm_mom6_cice5_cfsr.IN:24]: I think you also need 2 more changes: 1) add test dependency in rt.conf 2) copy the cold start test mediator restart files to NEW_BASELINE directory when the cold start finishes.

Comment (binli2337) [reply tests/datm_conf/cpld_datm_mom6_cice5_cfsr.IN:24]: Right now, the mediator files from cold-start runs are copied to NEW_BASELINE directory after the cold-start runs finish.  The dependency feature will be added.

Comment (junwang-noaa) [reply tests/datm_conf/cpld_datm_mom6_cice5_cfsr.IN:24]: Thanks, Bin. This works.

**Review by DeniseWorthen — APPROVED** [review]

**Pull request merged**